### PR TITLE
Compatibility guide for Kotlin 1.7.20

### PR DESCRIPTION
**Pull request by ilya-g**

### CODE REVIEW

File: docs/topics/compatibility-guide-1720.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+- _behavioral_: a change is said to be behavioral-incompatible if the same program demonstrates different behavior
+  before and after applying the change
+
+Remember that those definitions are given only for pure Kotlin. Compatibility of Kotlin code from the other languages
+perspective
+(for example, from Java) is out of the scope of this document.
+
+## Language
+
+<!--
```
Comment (PetrakovichVictoria) [inline docs/topics/compatibility-guide-1720.md:29]: Is sth still planned to be added here?

File: docs/topics/compatibility-guide-1720.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+> **Issue**: [KT-53797](https://youtrack.jetbrains.com/issue/KT-53797)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin 1.7 introduced a feature called unrestricted builder inference, so that even the lambdas passed
+> to parameters not annotated with `@BuilderInference` could benefit from the builder inference. 
+> However, that can cause several problems if more than one such lambda occurred in a function invocation.
```
Comment (PetrakovichVictoria) [inline docs/topics/compatibility-guide-1720.md:73]:
```suggestion
> However, that could cause several problems if more than one such lambda occurred in a function invocation. 
```

**Comment by PetrakovichVictoria** [comment]
Everything else except that minor comment is good. Approve.

**Comment by ilya-g** [comment]
@PetrakovichVictoria Thanks. Consider mentioning this compatibility guide in "What's new".

**Pull request merged**